internal/flink/command_statement_list.go

```diff
@@ -25,7 +25,13 @@ func (c *command) newStatementListCommand() *cobra.Command {
 }
 
 func (c *command) statementList(cmd *cobra.Command, args []string) error {
-	client, err := c.GetFlinkGatewayClient(false)
+	computePoolOnly := false
+
+	if c.Context.GetCurrentFlinkComputePool() != "" {
+		computePoolOnly = true
+	}
```

Comment:
This logic seems weird. What if all three flags, `--cloud`, `--region`, and `--compute-pool` are set? It looks like we'll use `--compute-pool` (which was originally designed to be a filter flag) and ignore `--cloud` and `--region`...

Reply:
You're right, we'll use the --compute-pool, but that's imo also better UX for the user because it prevents him getting empty results if he has a typo and starts wondering why (happened to our engineers yesterday and that's why I created the PR to make the UX for the command a bit better). 

If the user provides a compute-pool he wants to use for filtering and we didn't fetch the cloud/region from the compute pool, he *had* to also provide exactly the right provider/region or else he would get empty results and wonder why.

Reply:
Maybe we should be explicitly checking `--compute-pool` instead of using `c.Context.GetCurrentFlinkComputePool()`, then. If the user does `confluent flink compute-pool use` followed by `confluent flink statement list --cloud <> --region <>` it'll actually use the set compute pool and ignore the cloud and region that are passed in.

Reply:
Makes sense! Feel free to move on and do it 👍